compiler/src/Transform.js

```diff
@@ -58,10 +58,31 @@ function Transform(
                 "Constructing Transform: transformFunc must take parameters (whose names match the dbsource output)"
             );
 
-        this.columnNames = matches
-            .join(",")
-            .replace(/\/\/ *@result: */g, "")
-            .split(","); // safe bec we restricted the charset above
+        if (Array.isArray(columnNames)) {
+            this.columnNames = matches
+                .join(",")
+                .replace(/\/\/ *@result: */g, "")
+                .split(","); // safe bec we restricted the charset above
+            numColumns = columnNames.length;
+            this.allowUpdates = false;
+        } else if (typeof columnNames == "object") {
+            this.columnNames = Object.keys(columnNames);
+            numColumns = this.columnNames.length;
+            this.reverseFunctions = {};
+            for (let i = 0; i < numColumns; i++) {
+                let colName = this.columnNames[i];
+                if (columnNames[colName] !== null) {
+                    let funcBody = columnNames[colName].toString();
+                    funcBody = "return " + funcBody;
+                    this.reverseFunctions[colName] = funcBody;
+                }
+            }
+            this.allowUpdates = true;
+        } else {
+            throw new Error(
+                "Constructing Transform: columnNames must be either an Array of strings or an Object mapping string -> function"
+            );
+        }
```

Comment:
The object way of specifying a transform is still "experimental", so don't worry to support it. You can remove these for now.